alacritty_terminal/src/ansi.rs

```diff
@@ -25,62 +25,49 @@ use vte;
 use crate::term::color::Rgb;
 
 // Parse color arguments
-//
-// Expect that color argument looks like "rgb:xx/xx/xx" or "#xxxxxx"
-fn parse_rgb_color(color: &[u8]) -> Option<Rgb> {
-    let mut iter = color.iter();
 
-    macro_rules! next {
-        () => {
-            iter.next().map(|v| *v as char)
-        };
+fn is_rgb_separator(ch: Option<char>) -> bool {
+    ch == None || ch == Some('/') || ch == Some('\x07')
+}
+
+// Expect that color argument looks like "rgb:x[xxx]/x[xxx]/x[xxx]" or "#xxxxxx"
+fn parse_rgb_color(color: &[u8]) -> Option<Rgb> {
+    let len = color.len();
+    // Legacy colors are in the form "#xxxxxx"
+    // The use of this format is no longer encouraged by xparsecolor
+    let mut is_legacy = false;
+
+    if len >= 2 && color[0] == b'#' {
+        is_legacy = true;
+    } else if len < 9 || &color[..4] != b"rgb:" {
+        return None;
     }
 
-    macro_rules! parse_hex {
-        () => {{
-            let mut digit: u8 = 0;
-            let next = next!().and_then(|v| v.to_digit(16));
-            if let Some(value) = next {
-                digit = value as u8;
-            }
+    let mut iter = color.iter().map(|v| *v as char).skip(if is_legacy { 1 } else { 4 });
+    let mut colors = [0; 3];
 
-            let next = next!().and_then(|v| v.to_digit(16));
-            if let Some(value) = next {
-                digit <<= 4;
-                digit += value as u8;
-            }
-            digit
-        }};
-    }
+    for color in &mut colors {
+        *color = iter.next().and_then(|v| v.to_digit(16))? as u8;
+        *color <<= 4;
 
-    match next!() {
-        Some('r') => {
-            if next!() != Some('g') {
-                return None;
-            }
-            if next!() != Some('b') {
-                return None;
-            }
-            if next!() != Some(':') {
-                return None;
-            }
+        let next = iter.next();
 
-            let r = parse_hex!();
-            let val = next!();
-            if val != Some('/') {
-                return None;
-            }
-            let g = parse_hex!();
-            if next!() != Some('/') {
-                return None;
-            }
-            let b = parse_hex!();
+        if !is_legacy && is_rgb_separator(next) {
+            continue;
+        }
 
-            Some(Rgb { r, g, b })
-        },
-        Some('#') => Some(Rgb { r: parse_hex!(), g: parse_hex!(), b: parse_hex!() }),
-        _ => None,
+        *color += next.and_then(|v| v.to_digit(16))? as u8;
+
+        if is_legacy {
+            continue;
+        }
+
+        if (0..3).find(|_| is_rgb_separator(iter.next())) == None {
+            return None;
+        }
```

Comment:
We should not just skip it if colors have additional precision. XParseColors has very clear instructions on how additional depth should be handled.

Reply:
I'm sorry, but I can't find any references of what to do with the additional bit depth. If you could point me in the right direction I would apprecitae it.